tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -750,3 +750,85 @@ def func(x):
     results2 = func2(param)
     for r1, r2 in zip(results1.keys(), results2.keys()):
         assert r1 == r2
+
+
+@pytest.mark.torch
+@pytest.mark.parametrize("shots", [4000, [4000, 4001]])
+@pytest.mark.parametrize("postselect", [None, 1])
+@pytest.mark.parametrize("diff_method", [None, "best"])
+@pytest.mark.parametrize("measure_f", [qml.counts, qml.probs, qml.sample, qml.expval, qml.var])
+@pytest.mark.parametrize("meas_obj", [qml.PauliX(1), [0], [0, 1], "composite_mcm", "mcm_list"])
+def test_torch_integration(shots, postselect, diff_method, measure_f, meas_obj):
+    """Test that native MCM circuits are executed correctly with Torch"""
+    if measure_f in (qml.var, qml.expval) and (
+        isinstance(meas_obj, list) or meas_obj == "mcm_list"
+    ):
+        pytest.skip("Can't use wires/mcm lists with var or expval")
+
+    import torch
+
+    dev = get_device(shots=shots)
+    param = torch.tensor(np.pi / 3)
+
+    @qml.qnode(dev, diff_method=diff_method)
+    def func(x):
+        qml.RX(x, 0)
+        m0 = qml.measure(0)
+        qml.RX(0.5 * x, 1)
+        m1 = qml.measure(1, postselect=postselect)
+        qml.cond((m0 + m1) == 2, qml.RY)(2.0 * x, 0)
+        m2 = qml.measure(0)
+
+        mid_measure = 0.5 * m2 if meas_obj == "composite_mcm" else [m1, m2]
+        measurement_key = "wires" if isinstance(meas_obj, list) else "op"
+        measurement_value = mid_measure if isinstance(meas_obj, str) else meas_obj
+        return measure_f(**{measurement_key: measurement_value})
```

Comment:
I'm okay with leaving the other tests as they are, since it is much easier to read and understand the tests in their current state. If we start parametrizing more, I'm concerned about making the tests harder to parse.